### ISSUE TITLE
Hygiene, please

### ISSUE THREAD

**Issue by retronym** [scala/pickling#329]
I was just casually reviewing the pickler macro as I was referring someone to it as an example of a battle tested implementation.

I spotted a few bugs here:

https://github.com/scala/pickling/blob/v0.10.0-M4/core/src/main/scala/pickling/Macros.scala#L57-L59

The identifiers `FastTypeTag`, `scala`, `Unit`, `Any`, `String` should be fully qualified. I'd suggest to review other spots in the code for similar problems.

A good way to test this is to test it with something along the lines of:

``` scala
class Test {
  def hygiene() {
    val scala, Any, String, FastTypeTag, Unit = ()
    trait scala; trait Any; trait String; trait FastTypeTag; trait Unit
    List(Map(1 -> (Some(true) ())).pickle.unpickle
  }
}
```

**Comment by retronym** [comment]
I've just submitted a patch to Play for the same sort of thing: https://github.com/playframework/playframework/pull/4369

**Comment by jsuereth** [comment]
Thanks @retronym.  I was aware of a lot of hygiene issues we just didn't have time to fix before 0.10.0.  We'll try to prioritize soon.

**Comment by retronym** [comment]
Fixed by #340
